include/target-utils/rbt/rbt.h

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright(c) 2024 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdint.h>
+#include <stdbool.h>
+
+#define BLACK 0
+#define RED   1
+
+typedef enum bf_rbt_sts_t {
+  BF_RBT_OK,
+  BF_RBT_ERR = 3,
+  BF_RBT_NO_KEY,
+  BF_RBT_KEY_EXISTS
+} bf_rbt_sts_t;
+
+enum rbt_node_direction {
+	left_node,
```

Comment:
Maybe in uppercase? Otherwise they look like variables when used. Maybe `BF_RBT_LEFT_NODE` ?

Reply:
That's a good suggestion, will update